loadsave.c

```diff
@@ -50,8 +54,23 @@ create_directory_listing(uint8_t *data)
 	*data++ = 'P';
 	*data++ = 'C';
 	*data++ = 0;
+    
+    char filename[256];
+    uint8_t len = MIN(RAM[FNLEN], sizeof(filename) - 1);
+    memcpy(filename, (char *)&RAM[RAM[FNADR] | RAM[FNADR + 1] << 8], len);
+    filename[len] = 0;
+    
+    //use correct file location for ios
+    #if __APPLE__ && (TARGET_IPHONE_SIMULATOR || TARGET_OS_IPHONE)
```

Comment:
Now you're overwriting filename in all cases. The copying above had no effect.

(Seems correct, since the filename should be "$[...]" in this case, and can't be used as a filename/path on the host.)